app/code/Magento/MediaGallerySynchronization/Model/Filesystem/FileInfo.php

```diff
@@ -0,0 +1,296 @@
+        string $filename,
+        string $extension,
+        string $basename,
+        string $pathname,
+        int $perms,
+        int $inode,
+        int $size,
+        int $owner,
+        int $group,
+        int $aTime,
+        int $mTime,
+        int $cTime,
+        string $type,
+        $realPath,
+        \SplFileInfo $fileInfo,
+        \SplFileInfo $pathInfo
+    ) {
+        parent::__construct($file_name);
+        $this->path = $path;
+        $this->filename = $filename;
+        $this->extension = $extension;
+        $this->basename = $basename;
+        $this->pathname = $pathname;
+        $this->perms = $perms;
+        $this->inode = $inode;
+        $this->size = $size;
+        $this->owner = $owner;
+        $this->group = $group;
+        $this->aTime = $aTime;
+        $this->mTime = $mTime;
+        $this->cTime = $cTime;
+        $this->type = $type;
+        $this->realPath = $realPath;
+        $this->fileInfo = $fileInfo;
+        $this->pathInfo = $pathInfo;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPath(): string
+    {
+        return $this->path;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getExtension(): string
+    {
+        return $this->extension;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBasename($suffix = null): string
+    {
+        return $this->basename;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPathname(): string
+    {
+        return $this->pathname;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPerms(): int
+    {
+        return $this->perms;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getInode(): int
+    {
+        return $this->inode;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSize(): int
+    {
+        return $this->size;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOwner(): int
+    {
+        return $this->owner;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getGroup(): int
+    {
+        return $this->group;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getATime(): int
+    {
+        return $this->aTime;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getMTime(): int
+    {
+        return $this->mTime;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getCTime(): int
+    {
+        return $this->cTime;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getRealPath()
+    {
+        return $this->realPath;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getFileInfo($class_name = null): \SplFileInfo
```

Comment:
Please remove the functions that are not currently needed (additional functions can be added to this class later if they will be required). Also please remove the `\SplFileInfo` references from this class